### PR TITLE
allow `is` and `is not` for type comparisons (E721)

### LINKED ISSUE #830: E721: Naive for modules named `types`
This relates to #671 

## Install Method and Version
```
pip install --user pycodestyle
pycodestyle --version
2.4.0
```

## Context

E721 is the error for comparing the `type(x)` of something with another instead of using `isinstance`. Looking at #671, I thought this was something that I could submit a PR for. [Looking at the testsuite for E72, however](https://github.com/PyCQA/pycodestyle/blob/master/testsuite/E72.py#L15,L16), this is expected behavior currently.

## Problem

Any object named `types` is assumed to be the stdlib `types` module. There are many just reasons for a library to have a `types` module making selective disabling of E721 mandatory. (or alias imports).

```python

from foo import types

some_value = get_column_from_db()
if some_value == types.MyThing.SOME_PROPERTY:
    ...

```

Submitting an issue instead of PR because I'm not sure what context can be available to checks. :) The goal of the project is to be quick so the `ast` is out of the question. With that, either:

* The scope of the check should be reduced to not include `types` module checking
  * If we can't do it right, we shouldn't settle for "good enough" for an E-level check
* Regex could be expanded to include known contents of `types`
  * This still seems off because more maintenance

### LINKED ISSUE #1084: E721 regression(s)
so #1041 brought up a lot more instances of E721 and I don't feel confident adding this in a release since it's going to cause a bunch of (imo) unnecessary work

for example, here's something which wasn't flagged before but is now:

```python
assert type(x) is not bytes
```

strictly speaking the lint rule probably _should_ match this, but almost every case I can find with `is` and `is not` is being very explicit about the types being checked

I think the most reasonable thing to do here is to only check `==` and `!=` -- those are the ones that are most likely actual mistakes rather than an explicit check.

the other option is revert #1041 -- which might be the quickest thing to do to not block a release -- unclear what the best approach is here

### PR DESCRIPTION
**Pull request by asottile**
see #1084 and #1041 for more details

resolves #1084 
resolves #830

### CODE REVIEW

File: pycodestyle.py [hunk 1]
```diff
@@ -134,8 +134,10 @@
                                      r'\s*(?(1)|(None|False|True))\b')
 COMPARE_NEGATIVE_REGEX = re.compile(r'\b(?<!is\s)(not)\s+[^][)(}{ ]+\s+'
                                     r'(in|is)\s')
-COMPARE_TYPE_REGEX = re.compile(r'(?:[=!]=|is(?:\s+not)?)\s+type(?:s.\w+Type'
```
Comment (FichteFoll) [inline pycodestyle.py:137]: I can't find the rationale for removing `types.*Type` comparisons in the linked issues. As far as I can tell, those should still be included in the checks, shouldn't they?

Comment (asottile) [reply pycodestyle.py:137]: I think those got removed because the name `types` isn't necessarily the stdlib `types` module -- not 100% sure though

Comment (asottile) [reply pycodestyle.py:137]: the original patch is here: https://github.com/PyCQA/pycodestyle/pull/1041 -- this is just a revert-revert of that with an additional commit on top

Comment (FichteFoll) [reply pycodestyle.py:137]: I suppose the precise reasoning is the following, then:

> This check also breaks many ligimitate cases (such as having your own types module with an Enum class for instance, and comparing a value to the enum member).

which I can relate to. You can, of course, also shadow the built-in `type` function, but that's just not something people do in general and having a custom `types` module is much more likely, although also ending your enum in `Type` is most likely less common.

**Comment by asfaltboy** [comment]
@asottile my apologies for breaking the valid use cases in #1041, and thank you for following up with this fix and addressing the nuisance. 🙇

**Pull request merged**